File: app-mvc-kotlin/src/main/java/com/naveed/samples/apputils/events/ListDataEvent.java
package com.naveed.samples.apputils.events;

import java.util.List;

/**
 * Created by naveedali on 9/9/17.
 */

public class ListDataEvent<T> implements BaseEvent {

    private SimpleEvent simpleEvent;
    public List<T> listData;

    public ListDataEvent(boolean pStatus, int pId, String pMessage) {
        simpleEvent = new SimpleEvent(pStatus, pId, pMessage);
    }

    public ListDataEvent(boolean pStatus, int pId, String pMessage, List<T> listData) {
        this(pStatus, pId, pMessage);
        this.listData = listData;

    }

    @Override
    public boolean getStatus() {
        return simpleEvent.status;
    }

    @Override
    public int getEventId() {
        return simpleEvent.id;
    }

    @Override
    public String getMessage() {
        return simpleEvent.message;
    }


}

File: app/src/main/java/com/naveed/samples/apputils/events/ListDataEvent.kt
package com.naveed.samples.apputils.events

/**
 * Created by naveedali on 9/9/17.
 */

class ListDataEvent<T>(pStatus: Boolean, pId: Int, pMessage: String) : BaseEvent {

    private val simpleEvent: SimpleEvent = SimpleEvent(pStatus, pId, pMessage)
    lateinit var listData: List<T>

    override val status: Boolean
        get() = simpleEvent.status

    override val eventId: Int
        get() = simpleEvent.eventId

    override val message: String
        get() = simpleEvent.message

    constructor(pStatus: Boolean, pId: Int, pMessage: String, listData: List<T>) : this(pStatus, pId, pMessage) {
        this.listData = listData

    }


}
